# Welcome to the BITMaB workshop - Corpus Christi, TX (January 9-13, 2017)

This workshop provides training on how to describe the taxonomic diversity in a marine benthic community, using both morphological taxonomy and environmental sequencing approaches.

Final workshop Schedule is here: https://docs.google.com/document/d/1H8aHrddJ6VoBwrK6XWA9Zqs5fj8IkqbZIfniPwMKEUQ/edit?usp=sharing

Main Dropbox Folder with all workshop resources is available here: https://www.dropbox.com/sh/p6udtygnk7cf8dr/AAAmpar-QO7jCF_RRxnXigAia?dl=0 

Participant List can be found here: 
https://www.dropbox.com/s/ov3gr2q1k11wl9z/BITMaB_participants_list_2017.pdf?dl=0)

####Sample Info:

Google Map of Workshop Sample Sites:
https://drive.google.com/open?id=1PySPGX4VuOVBa2Hq9qdse7c88xU&usp=sharing

QIIME-formatted sample mapping file (Genomic Samples - Ethanol and Zymo Buffer):
INSERT LINK

---

# Workshop Schedule (permanent links to talks & handouts)

#### Monday Jan 9th:

* [Introduction to Genomics, Bioinformatics: DNA sequencing and Assembly of Genomes](https://www.dropbox.com/s/cds8uj2l81mounz/KelleyThomas-Intro2NGS-9Jan2017.pdf?dl=0) (Kelley Thomas)
    * [Video links to tutorial presented in the BITMaB Workshop can be found here (New Hampshire INBRE website)](http://nhinbre.org/bioinformatics-modules/)
* [Introduction to the command line environment and the world of BASH](NEED LINK) (Jordan Ramsdell)
* **Taxonomy Lecture**: [Goals of the workshop and how it fits into the context of the research project “Genomic Responses to the Deepwater Horizon event and development of high-throughput biological assays for oil spills”](https://www.dropbox.com/s/krr1xn3x814d0nr/FrancescoLeasi-WorkshopGoals-9Jan2017.pdf?dl=0) (Francesca Leasi)
* **Taxonomy Lecture**: [Harpacticoid Copepods](https://www.dropbox.com/s/jpy1o1iwmrej4yt/MelissaRohal-Harpacticoida-9Jan2017.pdf?dl=0) (Melissa Rohal)

#### Tuesday Jan 10th:

* Bash and Command line tricks, continued  (Jordan Ramsdell and UNH team)
* [Data analysis in the QIIME pipeline - first half of Slide Deck](https://www.dropbox.com/s/07k5ve9gehfwi9j/QIIME-workshop-slides-HBik.pdf?dl=0) (Holly Bik)
* **Taxonomy Lecture**: [Sipuncula](https://www.dropbox.com/s/iuk69djfave8y3s/AnjaSchulze-Sipuncula-10Jan2017.pdf?dl=0) (Anja Schulze)
* **Taxonomy Lecture**: [Gastrotricha](https://www.dropbox.com/s/vedfgjjebigjqc6/RickHochberg-Gastrotricha-10Jan2017.pdf?dl=0) (Rick Hochberg)

#### Wednesday Jan 11th:

* [Data analysis in the QIIME pipeline - jump to second half of Slide Deck ("Day 2")](https://www.dropbox.com/s/07k5ve9gehfwi9j/QIIME-workshop-slides-HBik.pdf?dl=0) (Holly Bik)
    * [Data analysis in the QIIME pipeline - Github Tutorial](https://github.com/BikLab/BITMaB-workshop/blob/master/QIIME-metabarcoding-tutorial.md)
    * [QIIME tutorial for already-demultiplexed raw data](https://github.com/BikLab/BITMaB-workshop/blob/master/QIIME-metabarcoding-tutorial-already-demultiplexed-fastqs.md) (afternoon session focused on custom analyses for people who brough data)
* **Taxonomy Lecture**: [Platyhelminthes](https://www.dropbox.com/s/c6t4r87sundpo3a/JulianSmith-Platyhelminthes-11Jan2017.pdf?dl=0) (Julian Smith III)
* **Taxonomy Lecture**: Nemertea (Jon Norenburg)

#### Thursday Jan 12th:

* Genome assembly and annotation (Jordan Ramsdell and UNH team)
* Introduction to Whole Metagenome Shotgun Analysis.  (Kelley Thomas)
* Mapping DNA sequence reads to reference genomes to investigate genetic diversity or profiling gene expression. (Kelley Thomas)
* **Taxonomy Lecture**: [Aplacophora](https://www.dropbox.com/s/k6pb4vr8uiunbva/KevinKocot-Aplacophora-12Jan2017.pdf?dl=0) (Kevin Kocot)
    * Related paper that Kevin mentioned during his talk: [https://academic.oup.com/sysbio/article/doi/10.1093/sysbio/syw079/2449704/Phylogenomics-of-Lophotrochozoa-with-consideration](https://academic.oup.com/sysbio/article/doi/10.1093/sysbio/syw079/2449704/Phylogenomics-of-Lophotrochozoa-with-consideration)
* **Taxonomy Lecture**: Kinorhyncha, Tardigrada, Gnathostomulida, and Loricifera (Martin Sørensen)

#### Friday Jan 13th:

* Visualizing metabarcoding data, and other data analysis tools outside of QIIME (Holly Bik)
    * [Links to LEfSE, Phinch, Phyloseq listed on GitHub tutorial (end section - "Other Resources")](https://github.com/BikLab/BITMaB-workshop/blob/master/QIIME-metabarcoding-tutorial.md)
    *  Other examples of useful data visualization in metabarcoding studies:
        * [Figures in this recent ATM Keypad paper (Bik et al. 2016)](http://msphere.asm.org/content/1/6/e00226-16)
        * [Figures in GOM Deepwater Horizon Impacts paper (Bik et al. 2012)](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0038550)
        * [Illumina run artifact visualizations/exploration in this Office Microbiome paper (Chase et al. 2016)](http://msystems.asm.org/content/1/2/e00022-16)
* [Reccomendations for Enhancing Research Reproducibility](https://www.dropbox.com/s/sxc0o3a83ziu52k/FASEB_Enhancing%20Research%20Reproducibility%202016.pdf?dl=0) (Kelley Thomas)  
* **Taxonomy Lecture**: [Nematoda](https://www.dropbox.com/s/47ktyehy51fz40m/AlbertodeJesusNavarrete-Nematoda-13Jan2017.pdf?dl=0) (Alberto de Jesús Navarrete)
* **Taxonomy Lecture**: [Ostracoda](https://www.dropbox.com/s/qw0ybv93j0lmma6/SimoneBrand%C3%A3o-Ostracoda-13Jan2017.pdf?dl=0) (Simone Nunes Brandão)






---
# Getting started
### Login using the username assigned to you

```
ssh username@ron.sr.unh.edu
```
### Setting up a temp directory on Ron
This setup is specific to Ron and may not be needed for other servers.

* **Create a temp directory in your home directory**

* **Open your `~/.bashrc` in nano**

* **Type the following line at the end of your `~/.bashrc` and save the file.**

```
export TMPDIR=/home/gomre/USERNAME/temp
```

* **Source your `~/.bashrc`**

### To use Qiime, run the commands below

```
enable_qiime.sh
```
```
source activate qiime1
```

Now, you should be in the QIIME environment and ready to run through the pipeline. 

***

### Where are all the files?
The locations of all the files you will need for the QIIME tutorial are listed below

**Data file** - GOM Illumina seqeunces, in fasta format:

```
/home/gomre/taruna/200k-GOM-Illumina-subsample/data-clean/GOM_concat1.7_subsample200k_10Jan14.fna
```

**Mapping file** - tab-delimited spreadsheet containing sample metadata:

```
/home/gomre/taruna/200k-GOM-Illumina-subsample/qiime-files/mapping-files/QIIMEmappingfile_GOM_Illumina_7Apr14_F04.txt
```
**Parameters file** - 

```
/home/gomre/taruna/200k-GOM-Illumina-subsample/qiime-files/parameters/18S_openref99_rdp_silva119.txt
```

**Silva119 database** - Publically available reference database files, [downloaded from the QIIME website](http://qiime.org/home_static/dataFiles.html). SILVA is the normal database we use for microbial eukayote metabarcoding analyses (meiofauna, protists, etc.), but other databases you can use include Greengenes (for bacterial/archaeal 16S rRNA data) and UNITE (for fungal ITS rRNA data). Both of these are also available for download at the above QIIME link. Or you can build and use your own custom database! 

**18S rep set file** - unaligned reference sequences, in fasta format

```
/home/gomre/taruna/200k-GOM-Illumina-subsample/ref-dbs/Silva119_release/rep_set_eukaryotes/99/Silva_119_rep_set99_18S.fna
```

**Aligned 18S rep set fasta** - the reference SILVA sequences which were subjected to multiple sequence alignment (using secondary structure information from rRNA)

```
/home/gomre/taruna/200k-GOM-Illumina-subsample/ref-dbs/Silva119_release/aligned_rep_files/99_18S_only/Silva_119_rep_set99_aligned_18S_only.fna
```

**Taxonomy file** - mapping file which links sequence IDs to taxonomic hierarchies (what we usually refer to as "taxonomy strings"")

```
/home/gomre/taruna/200k-GOM-Illumina-subsample/ref-dbs/Silva119_release/consensus_majority_taxonomy/consensus_taxonomy_eukaryotes/99/taxonomy_99_7_levels_consensus.txt
```
Full directory of SILVA database files (For reference, in case you want to look around or try out some customized analyses)

```
/home/gomre/taruna/200k-GOM-Illumina-subsample/ref-dbs/Silva119_release/
```